src/lib/weather-data.component.ts

```diff
@@ -1,73 +1,431 @@
+     */
+    setFilterMap(): void {
+        this.activatedFilters.forEach((item) => {
+            this.filterStates.set(item, true);
+        });
+    }
 
-  weatherData: any;
+    /**
+     * get all station data from dwd.
+     * safe them in stations
+     */
+    getAllStations(): void {
+        this.weatherService.fetchStations("/").subscribe((stationData) => {
+            this.stations = stationData;
+            this.showStations();
+        });
+    }
 
-  possibleResolutions!: any;
+    //------------------------------------------------------------------------------ Filter Stations on Map -------------------------------------------
 
-  constructor(public weatherService: WeatherDataService) { }
+    /**
+     * filter list of stations to fit criteria and initiate marker creation
+     */
+    showStations(): void {
+        let filteredStations: Station[] = this.stations;
 
-  ngOnInit(): void {
-    this.getStationsforMap();
-  }
+        // show only stations which have recent data (until today)
+        if (this.historicalFiltered) {
+            filteredStations = filteredStations.filter((station) => {
+                return !station.historical;
+            });
+        }
+        // show only stations which don't have recent data
+        else {
+            filteredStations = filteredStations.filter((station) => {
+                return station.historical;
+            });
+        }
+
+        this.filterStates.forEach((filterValue, filterKey) => {
+            if (filterValue) {
+                filteredStations = this.filterStationsByDataType(filteredStations, filterKey);
+            }
+        });
+
+        this.createStationMarkers(filteredStations);
+    }
+
+    /**
+     * switches filterOption in filterStates map to reflect which filters are active
+     * @param filterOption dataCapability to switch state of filtering
+     */
+    switchFilterOption(filterOption: DataCapability): void {
+        let bool = this.filterStates.get(filterOption);
+        this.filterStates.set(filterOption, !bool);
+    }
+
+    /**
+     * searches an array of stations for every station which holds a certain datatype in their
+     * capabilities attribute
+     * @param stations array to search in
+     * @param dataTypeToFilter to search for
+     * @returns array of stations with the searched datatype
+     */
+    filterStationsByDataType(stations: Station[], dataTypeToFilter: string): Station[] {
+        return stations.filter((station) =>
+            station.capabilities.some((capability) => capability.dataType === dataTypeToFilter)
+        );
+    }
+
+    /**
+     * set historical filter flag
+     */
+    filterHistorical(): void {
+        this.historicalFiltered = !this.historicalFiltered;
+    }
+
+    //------------------------------------------------------------------------------ Markers and Tooltips ---------------------------------------------
+
+    /**
+     * create map markers based on stations
+     */
```

Comment:
the parameter description is missing here
```suggestion
     * create map markers based on stations
     * @param stations array of stations for marker creation
     */
```